app/services/CheckoutService.scala

```diff
@@ -0,0 +1,28 @@
+package services
+
+import com.gu.identity.play.IdMinimalUser
+import com.typesafe.scalalogging.LazyLogging
+import model.SubscriptionData
+import play.api.mvc.Request
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class CheckoutService(identityService: IdentityService) extends LazyLogging {
+
+  def processSubscription(subscriptionData: SubscriptionData, request: Request[_]): Future[Either[GuestUserNotCreated, IdMinimalUser]] =  {
```

Comment:
Think this comment got lost before- rather than pass through the `request` here, perhaps better to call `AuthenticationService.authenticatedUserFor(request)` in the controller, and just pass thru an `Option[IdMinimalUser]`?